src/checkbox-group.ts

```diff
@@ -467,21 +424,27 @@ export default class CheckboxGroup extends LitElement implements FormControl {
   }
 
   @state()
-  private isBlurring = false;
+  private hasReportedValidityOrSubmitted = false;
 
   @state()
-  private isCheckingValidity = false;
+  private isBlurring = false;
 
   @state()
-  private isReportValidityOrSubmit = false;
+  private isCheckingValidity = false;
 
   @state()
   private validityMessage?: string;
 
+  #checkedAndEnabledCheckboxes = new Set<Checkbox>();
+
   #componentElementRef = createRef<HTMLElement>();
 
   #defaultSlotElementRef = createRef<HTMLSlotElement>();
 
+  #focusedCheckbox: Checkbox | null = null;
```

Comment:
> Checkbox Group no longer moves focus to its first enabled Checkbox on submit when another Checkbox has focus. When no Checkbox is focused, Checkbox Group still moves focus to its first enabled Checkbox.

To support the above fix, Checkbox Group has to track which Checkbox, if any, has focus.